packages/cli/src/util/deploy/process-deployment.ts

```diff
@@ -281,3 +285,30 @@ export default async function processDeployment({
     throw err;
   }
 }
+
+export const archiveSuggestionText =
+  'Try using `--archive=tgz` to limit the amount of files you upload.';
+
+export class UploadErrorMissingArchive extends Error {
+  link = 'https://vercel.com/docs/cli/deploy#archive';
+}
+
+export function handleErrorSolvableWithArchive(error: unknown) {
+  const errorIsObject = typeof error === 'object' && error !== null;
+  if (errorIsObject && 'message' in error) {
+    const isUploadRateLimit =
+      'rateLimitName' in error &&
+      typeof error.rateLimitName === 'string' &&
+      ['api-upload-free', 'api-upload-paid', 'api-upload-extended'].includes(
+        error.rateLimitName
+      );
```

Comment:
This is getting a bit into "too much client-side logic" territory IMO. Maybe like this would be a little more future-proof?

```suggestion
      error.rateLimitName.startsWith('api-upload-')
```